src/components/Recipe/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { createPortal } from 'react-dom'
+
+import { fetchRecipe } from 'api'
+import Loader from 'components/common/Loader'
+import { Ingredient, RecipeDetails, Step } from 'types'
+
+import styles from './Recipe.module.scss'
+
+function Recipe() {
+  const { id } = useParams()
+
+  const [recipe, setRecipe] = useState<RecipeDetails>()
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+
+  useEffect(() => {
+    fetchRecipe(id!).then((response) => {
+      setRecipe(response.data)
+      setIsLoading(false)
+    })
+  }, [id])
+
+  return (
+    <>
+      {isLoading && createPortal(<Loader />, document.body)}
```

Comment:
I think should add createPortal directly in Loader, because everytime you will want to use Loader you will also need createPortal (DRY)